stdlib/nats/src/main/ballerina/nats/connection/connection.bal

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents a NATS connection.
+public type Connection object {
+
+    # Initializes a connection from the given server config.
+    #
+    # + config - Information neccasary for NATS client to establish a connection with the server.
+    public function __init(ConnectionConfig config) {
```

Comment:
Since `host` and `port` seem to be required fields in the `ConnectionConfig` record, should we maybe remove them from the record and make them positional params for `__init()`.

Reply:
There will be two main paradigms to use connection, 

a. We basically create a connection and reuse them between 1.* producers, 1.* consumers and combination of both of them (advance use case)
b. We introduce a simplified API to create a connection per producer or a consumer 

In order for us to fulfill both the aforementioned use cases, creation of the producer or the consumer should either accept an already established connection or accept the list of fields necessary to create a connection which is represented through ConnectionConfig and internally create a new connection  

If we extract the mandatory params included in the ConnectionConfig as positional parameters the downside would be the same list of parameters should be reflected in both producer and the consumer creation APIs to make it consistent and to ensure that API is more readable instead of having it flooded with too many parameters.

This would however be a subjective choice :)